pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/leap/LeapDriver.java

```diff
@@ -88,27 +93,40 @@ public String initialState() {
         bucket.asScheduler().consume(1);
         final List<PersistentQueueElement> events = new ArrayList<>();
         final HttpClient client = HttpClient.newHttpClient();
-        final String uri = apiUri + "/ClientApi/V1/DeviceReadings";
-        final HttpRequest request = HttpRequest.newBuilder()
-            .uri(URI.create(uri))
-            .header("Authorization", "Bearer " + getAuthToken())
-            .build();
+        final String uri;
+        if (state == "")
```

Comment:
This is not the correct way to check for an empty string. See https://stackoverflow.com/questions/513832/how-do-i-compare-strings-in-java. Replace with `state.isEmpty()`.

Reply:
Changed to ` state.isEmpty()` .